app/src/main/java/org/fossasia/openevent/general/event/EventService.kt

```diff
@@ -36,10 +38,14 @@ class EventService(private val eventApi: EventApi, private val eventDao: EventDa
     }
 
     fun getEventsByLocation(locationName: String): Single<List<Event>> {
-        return eventApi.searchEvents("name", locationName).map {
-            eventDao.insertEvents(it)
-            it
-        }
+        return eventApi.searchEvents("name", locationName)
+                .map {
+                    eventDao.insertEvents(it)
+                    it
+                }.flatMap {
+                    eventDao.getEventsWithIds(it.map { it.id })
```

Comment:
Wrong logic

Reply:
We are inserting all events returned by API into db and then retrieving the local copy of events. Isnt that what we need. Please guide me if any better way is there. Thank you

Reply:
This will load one ID and not a set of ID. Your rationale is correct, implementation is wrong

Reply:
`it.map{ it.id}` will return an Array with IDs of events fetched by the API and the DAO returns event from the database where IDs are in this Array
Tried running this locally events are getting fetched from the database (checked it since favorite drawable were automatically set)

Reply:
Checked PR locally, works fine :+1:

Reply:
OK, it may work, but the IDs should be mapped in a single query. This maps a single item from database and does `SELECT * from Event WHERE id in ?` 

This will make n queries where there can be one. As a proof, you may print something as debug statement and count how many times `getEventsWithIds` is called

That's what I meant to say

Reply:
okay so if I map and store Ids in an array above and then pass it to `getEventsWithIds` then it should do, I will check by logging and get back to you! Thanks @iamareebjamal

Reply:
No need to store in array. Use observable map and toList

Reply:
okay will do that

Reply:
@iamareebjamal by obervable map did you mean
 `Observable.fromIterable(it).map { it.id }.toList()` and pass this to dao
or can we just do
```
.flatMap {
   eventDao.getEventsWithIds(it.map { it.id }.toList())
}
```

Reply:
```Kotlin
.map {
  eventDao.insertEvents(it)
  it.id
}.toList()
.flatMap {
  // Now you have list ofn IDs
}
```

Reply:
`it` is a list of event we cant directly do `it.id`

Reply:
Then use flatMapIterable